data_diff/database.py

```diff
@@ -66,12 +70,77 @@ def _query_conn(conn, sql_code: str) -> list:
     return c.fetchall()
 
 
-class Database(ABC):
+class ColType:
+    pass
+
+
+@dataclass
+class PrecisionType(ColType):
+    precision: Optional[int]
+
+
+class TemporalType(PrecisionType):
+    pass
+
+
+class Timestamp(TemporalType):
+    pass
+
+
+class TimestampTZ(TemporalType):
+    pass
+
+
+class Datetime(TemporalType):
```

Comment:
Should we just _only_ do `Timestamp` and `TimestampTZ` for all of our representation to simplify? Do we really need `Datetime` too?

It seems most of the new databases who aren't too concerned with SQL standards just call everything a timestamp, and that the old distinction of `Timestamp` being about a current point-in-time and `Datetime` being for e.g. birthdays and stuff doesn't hold anymore.

Reply:
Datetime is just the database type, that we get from querying the schema. We already convert it to timestamp no matter what.

Reply:
Yeah, but then why have another class for it?

Reply:
Because we might want to make this distinction in the future, or add datetime/timestamp-specific behaviors. And saying "datetime" is Timestamp is just going to cause weird bugs for someone in the future.